platform/security/sts/security-sts-pkivalidator/src/main/java/org/codice/ddf/security/validator/pki/PKITokenValidator.java

```diff
@@ -261,8 +213,29 @@ public TokenValidatorResponse validateToken(TokenValidatorParameters tokenParame
             }
 
             Credential returnedCredential = validator.validate(credential, requestData);
-            response.setPrincipal(returnedCredential.getCertificates()[0].getSubjectX500Principal());
-            validateTarget.setPrincipal(returnedCredential.getCertificates()[0].getSubjectX500Principal());
+            X500Principal subjectX500Principal =
+                    returnedCredential.getCertificates()[0].getSubjectX500Principal();
+            response.setPrincipal(subjectX500Principal);
+            if (response.getAdditionalProperties() == null) {
+                response.setAdditionalProperties(new HashMap<>());
+            }
+            try {
+                String emailAddress = SubjectUtils.getEmailAddress(subjectX500Principal);
+                if (emailAddress != null) {
+                    response.getAdditionalProperties()
+                            .put("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress",
+                                    emailAddress);
+                }
+                String country = SubjectUtils.getCountry(subjectX500Principal);
```

Comment:
I'm just allowing someone to set a different claim identifier for each of the values. So if someone wants to have a claim of "PersonEmailAddress" instead of the long URI, then they can do that, instead of hardcoding it to always be the URI in every case.